CONTRIBUTING.md : Mac and Linux steps to start contribute

Tilix4 opened the pull request:
Mac and Linux steps added to explain how easily start contributing to the atom-fountain package.

jrbostic commented on CONTRIBUTING.md:
```diff
@@ -8,3 +8,17 @@ Logging an [issue](https://github.com/superlou/fountain-atom/issues) is a welcom
+`mkdir -p dev/packages` *Creates the folder tree for development*  
+`atom -d dev/packages/` *Go to the newly created folder*  
+`git clone https://github.com/YourPseudo/fountain-atom.git` *Clone your forked repository in the current folder*  
+`cd atom-fountain` *Go to the cloned folder*  
```
There's actually an atom command that can replace the above steps.  I'm not in a position to look it up ATM.  But it basically creates the directory for you and creates a symbolic link to your forked local copy.  Probably safer than cloning directly into the dev packages dir.

Tilix4 replied:
I looked around about that, and found the right steps to follow : https://flight-manual.atom.io/hacking-atom/sections/contributing-to-official-atom-packages/  
Then I modified my commit. I don't know if I did it correctly, I did `--amend` it and `push -f`.
It seems that the pull request is updated, but I'm still a noob in pull request procedure.

superlou commented:
I believe you shouldn't have to manually fork the repository. If you use `apm develop fountain` it should do all of the cloning/installing/linking automatically.

Tilix4 commented:
Okay, maybe I'm missing something. If I don't fork it, how do I manage my commit/push workflow ? How can I submit a Pull Request ?
And forking it it's what I read in the official documentation : 

> [For example, if you want to make changes to the tree-view package, fork the repo on your github account, then clone it](https://flight-manual.atom.io/hacking-atom/sections/contributing-to-official-atom-packages/#cloning)

jrbostic commented:
I think you're right that you do have to fork the repo to do a PR.  But the quickest possible workflow for hacking on a package is using the apm develop command... It clones the repo of the atom package locally into /github and symlinks it into dev packages.  Then if you fork the repo to your account you can set your fork as the remote and make superlou's repo the upstream, so you are able to pull changes from upstream but also able to push to your forked copy.  (For superlou it's enough to run the apm develop command because he has full rights on the main repo.)

superlou commented:
Agreed with @jrbostic. I like the `apm develop ...` approach then changing the upstream approach since it hides most of the Atom specific configuration.

Tilix4 commented:
Okay, I modified the CONTRIBUTING.md according to the `apm develop` procedure.

About `menus/fountain.cson` I didn't want to modify it in this pull request, I should have done some branch for that. I didn't know github will update with every commit I make. Now I'll be more careful.
Anyway, I added the Export PDF command in `Packages/Fountain`. I think it's useful to have it there if a new user doesn't know the shortcuts.
If you want to make it clean, please tell me what should I do.

jrbostic commented:
I like it.  The only thing about the process that I would note is that atom -d command opens atom in the current directory.  I personally prefer to run that from a specific directory where I store my fountain files.  If someone follows your directions exactly they would be opening the repo directory.  That's fine since there are some test files there...  But those files are also used for Jasmine tests, so if modified could cause failures.  Not a huge deal though.

I also think adding to the menu cson is a good idea.  As long as it provides the same messages and output (on success and error)... Like, if no file is selected, user gets the info message... Or if they're overwriting output PDF they are asked to confirm.

But at a glance it looks good to me!

Tilix4 commented:
ok thanks. I understand your worries about the workflow. But my description only explains how to quick start, not how to have a perfect workflow (mine is not even totally set). 

The Export PDF in the menu has exactly the same behaviour as the shortcut use. Then I think it works as expected !

jrbostic commented:
Sweet!

superlou commented:
In the future, let's split up stuff like the menu addition from the docs, but no reason to add extra work now. Ready to merge?

Tilix4 commented:
Totally !
